### PR TITLE
[FIX] correct the usage of `scipy.stats.gamma.pdf`

### LINKED ISSUE #3295: Possible inappropriate usage of scipy.stats.gamma.pdf in _gamma_difference_hrf
Here (`scipy.stats.gamma.pdf(x, a, loc=0, scale=1)`) is the gamma pdf function statement. And following is how it's used in nilearn. `delay / dispersion` and `dt / dispersion` strangely correspond to `a` and `loc` respectively, while `scale` with default value `1`.

https://github.com/nilearn/nilearn/blob/cc11c85cd222632dcaf7c51005b549f75a49bbc4/nilearn/glm/first_level/hemodynamic_models.py#L18-L74

See more details here #2981. With this usage, we cannot produce the right Glover HRF even if we add the normalization constant.

Nilearn version: 0.9.1

### Steps and code to reproduce bug

Here is the comparison between different versions of Glover HRF.

```python
import numpy as np
from scipy.stats import gamma
from nilearn.glm.first_level import glover_hrf
import matplotlib.pyplot as plt

# Glover 1999
def original_glover(t, n1=6.0, t1=0.9, a2=0.35, n2=12.0, t2=0.9):
    """As seen in Glover 1999, page 420. Using values for the auditory cortex (canonical Glover).
    Variables are named after those in the paper.
    """
    gamma1 = t**n1 * np.exp(-t / t1)
    gamma2 = t**n2 * np.exp(-t / t2)
    c1 = gamma1.max()
    c2 = gamma2.max()
    return gamma1 / c1 - a2 * gamma2 / c2


# glover_hrf with normalized gamma
def _gamma_difference_hrf(tr, oversampling=50, time_length=32., onset=0.,
                          delay=6, undershoot=16., dispersion=1.,
                          u_dispersion=1., ratio=0.167):
    """
    Minimum modification from nilearn.glm.first_level._gamma_difference_hrf
    (only add the normalization constant)
    """
    dt = tr / oversampling
    time_stamps = np.linspace(0, time_length,
                              np.rint(float(time_length) / dt).astype(int))
    time_stamps -= onset
    
    peak = gamma.pdf(time_stamps, delay / dispersion, dt / dispersion)
    undershoot = gamma.pdf(time_stamps, undershoot / u_dispersion, dt / u_dispersion)
    hrf = peak / peak.max() - ratio * undershoot / undershoot.max()
    hrf /= hrf.sum()
    return hrf


def glover_hrf_with_norm(tr, oversampling=50, time_length=32., onset=0.):
    """
    Same as nilearn.glm.first_level.glover_hrf
    """
    return _gamma_difference_hrf(tr, oversampling, time_length, onset,
                                 delay=6, undershoot=12., dispersion=.9,
                                 u_dispersion=.9, ratio=.35)


# code from dangom
# https://github.com/nilearn/nilearn/issues/2981#issuecomment-1160515684
def sample(t: np.array):
    # Glover parameters
    peak_delay: float = 6.0
    peak_width: float = 0.9
    undershoot_delay: float = 12.0
    undershoot_width: float = 0.9
    positive_negative_ratio: float = 0.35

    peak = gamma.pdf(t, peak_delay / peak_width, loc=1/50, scale=peak_width)
    undershoot = gamma.pdf(
        t,
        undershoot_delay / undershoot_width,
        loc=0,
        scale=undershoot_width,
    )
    peak_norm = peak.max()
    undershoot_norm = undershoot.max()

    hrf = peak / peak_norm - undershoot / undershoot_norm * positive_negative_ratio
    return hrf


t = np.arange(0, 32, 0.02)  # 32 seconds, with a TR=1 and oversampling=50 (hence 0.02).
# Plot both HRFs normalized so we can compare their shape.
plt.plot(t, original_glover(t) / original_glover(t).max(), label="Glover 1999")
plt.plot(t, glover_hrf(1) / glover_hrf(1).max(), label="nilearn glover_hrf")
plt.plot(t, glover_hrf_with_norm(1) / glover_hrf_with_norm(1).max(), label="glover_hrf with normalized gamma")
plt.plot(t, sample(t) / sample(t).max(), label="code from dangom")
plt.grid()
plt.legend()
plt.show()
```

Here is the result
image

### PR DESCRIPTION
**Pull request by const7**
Closes #3295.
Related to #2981.

Changes proposed in this pull request:

- Fix the usage of `scipy.stats.gamma.pdf`

### CODE REVIEW

**Comment by const7** [comment]
Thank you! I have added it.

And it seems there is a failed test induced from this https://github.com/nilearn/nilearn/commit/0f54d48138dd1fcb962815db39c6e395d5c0cfe4 commit. Is there anything else I should do?

**Review by bthirion — APPROVED** [review]
LGTM. I don't think that it's meaningful to add a new test selective for this change, so I'm happy to keep it as is.

**Review by ymzayek — COMMENTED** [review]
LGTM @const7! Can you fix the small typo below?

File: doc/changes/latest.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ Fixes
 - :func:`~surface.load_surf_mesh` loaded FreeSurfer specific surface files (e.g. `.pial`) with a shift in the coordinates. This is fixed by adding the c_ras coordinates to the mesh coordinates (:gh:`3235` by `Yasmin Mzayek`_).
 - Function ``nilearn.glm.second_level.second_level._check_second_level_input`` now raises an error when ``flm_object`` argument is ``False`` and ``second_level_input`` is a list of :class:`~glm.first_level.FirstLevelModel` (:gh:`3283` by `Matthieu Joulot`_).
 - Function :func:`~image.resample_img` now warns the user if the provided image has an ``sform`` code equal to 0 or None  (:gh:`3284` by `Matthieu Joulot`_).
+- Fix usage of ``scipy.stats.gamma.pdf`` in ``_gamma_difference_hrf`` function under ``nilearn/glm/first_level/homedynaic_models.py``, which resulted in slight distortion of HRF (:gh:`3297` by `Kun CHEN`_).
```
[the remark this answers is not in the document]
Comment (const7) [reply doc/changes/latest.rst]: Sure! Thanks for your kind reminder

**Comment by ymzayek** [comment]
Merging

**Pull request merged**